code-experiments/build/python/example_experiment_non_anytime.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
+"""An example experiment for benchmarking non-anytime optimization algorithms with restarts.
+
+Non-anytime algorithms are those whose parameters depend not only on the dimensionality of the problem, but also on the
+budget of evaluations. Benchmarking such algorithms therefore entails running experiments on a test suite using various
+(increasing) budgets. These budgets can/should be provided by the user. If the user does not specify a budget list, a
+default budget list is used.
+
+This code builds upon the example experiment for beginners and provides only simplistic progress information.
+
+To apply the code to a different solver, add an interface to function `fmin`.
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+import cocoex  # experimentation module
+from numpy.random import rand  # for randomised restarts
+
+# for postprocessing
+import cocopp
+import os
+import webbrowser
+
+# for solvers
+import scipy.optimize
+import cma
+
+
+def default_budget_list(max_budget=10, num=100):
+    """Produces a budget list with at most `num` different increasing budgets within [1, `max_budget`] that are equally
+    spaced in the logarithmic space.
+    """
+    from math import log10
```

Comment:
I believe that local imports are frowned upon in Python. I would tend to say rather
```python
import numpy as np
```
on top and living with 
```python
np.unique(np.logspace(0, np.log10(max_budget), num=num).astype(int))
```
is better style in an offline script.